src/sentry/search/events/builder/metrics.py

```diff
@@ -881,32 +893,56 @@ def run_query(self, referrer: str, use_cache: bool = False) -> Any:
+                                        spec=spec,
+                                        require_time_range=True,
+                                        groupby=[MetricGroupByField(field=c) for c in group_bys],
+                                    )
+                                )
                         else:
-                            metrics_query = transform_mqb_query_to_metrics_query(
-                                self.get_metrics_layer_snql_query(query_details, extra_conditions),
-                                self.is_alerts_query,
+                            metrics_queries.append(
+                                transform_mqb_query_to_metrics_query(
+                                    self.get_metrics_layer_snql_query(
+                                        query_details, extra_conditions
+                                    ),
+                                    self.is_alerts_query,
+                                )
+                            )
+                    metrics_data = []
+                    # XXX: Now metrics_data is a list
+                    for metrics_query in metrics_queries:
+                        with sentry_sdk.start_span(op="metric_layer", description="run_query"):
+                            metrics_data.append(
+                                get_series(
+                                    projects=self.params.projects,
+                                    metrics_query=metrics_query,
+                                    use_case_id=UseCaseID.TRANSACTIONS
+                                    if self.is_performance
+                                    else UseCaseID.SESSIONS,
+                                    include_meta=True,
+                                    tenant_ids=self.tenant_ids,
+                                )
                             )
-                    with sentry_sdk.start_span(op="metric_layer", description="run_query"):
-                        metrics_data = get_series(
-                            projects=self.params.projects,
-                            metrics_query=metrics_query,
-                            use_case_id=UseCaseID.TRANSACTIONS
-                            if self.is_performance
-                            else UseCaseID.SESSIONS,
-                            include_meta=True,
-                            tenant_ids=self.tenant_ids,
-                        )
                 except Exception as err:
+                    import traceback
+
+                    traceback.print_exc()
+
                     raise IncompatibleMetricsQuery(err)
                 with sentry_sdk.start_span(op="metric_layer", description="transform_results"):
-                    metric_layer_result = self.convert_metric_layer_result(metrics_data)
+                    # XXX: Until we can figure out how to handle a list
+                    metric_layer_result = self.convert_metric_layer_result(metrics_data[0])
```

Comment:
For now, I'm using the first element (`metrics_data[0]`) so we don't have to do any stitching of the data.